src/tests/resistance/fuzzing.rs

```diff
@@ -398,6 +820,143 @@ async fn fuzzing_slightly_corrupted_version_during_handshake_responder_side() {
     node.stop().await;
 }
 
+#[tokio::test]
+async fn fuzzing_slightly_corrupted_version_when_node_initiates_handshake() {
+    // ZG-RESISTANCE-003 (part 4, version only)
+    //
+    // This particular case is considered alone because it is at particular risk of causing
+    // troublesome behaviour, as seen with the valid metadata fuzzing against zebra.
+    //
+    // zebra: disconnects immediately.
+    // zcashd: Some messages get ignored and timeout. Most cause an immedietely due to
+    //          - main: PROCESSMESSAGE: INVALID MESSAGESTART, or
+    //          - net: Oversized message from peer
+
+    let locked_rng = Arc::new(Mutex::new(seeded_rng()));
+
+    // create tcp listeners for peer set (port is only assigned on tcp bind)
+    let mut listeners = Vec::with_capacity(ITERATIONS);
+    for _ in 0..ITERATIONS {
+        listeners.push(TcpListener::bind(new_local_addr()).await.unwrap());
+    }
+
+    // get list of peer addresses to pass to node
+    let peer_addresses = listeners
+        .iter()
+        .map(|listener| listener.local_addr().unwrap())
+        .collect::<Vec<_>>();
+
+    // start peer processes
+    let mut peer_handles = Vec::with_capacity(listeners.len());
+    for peer in listeners {
+        let peer_rng = locked_rng.clone();
+        peer_handles.push(tokio::time::timeout(
+            tokio::time::Duration::from_secs(120),
+            tokio::spawn(async move {
+                // Await connection and receive version
+                let (mut peer_stream, _) = peer.accept().await.unwrap();
+                let version = Message::read_from_stream(&mut peer_stream).await.unwrap();
+                assert_matches!(version, Message::Version(..));
+
+                // send bad version
+                let mut rng = peer_rng.lock_owned().await;
```

Comment:
Make sure you drop the lock as soon as it isn't needed anymore.

Reply:
I'll scope them - thanks